tests/root_gauge_factory_proxy/test_root_gauge_factory_proxy_admin.py

```diff
@@ -0,0 +1,275 @@
+import brownie
+from brownie import ETH_ADDRESS
+
+
+def test_set_manager(root_gauge_factory_proxy, alice, bob, charlie, chain):
+
+    default_ownership_admin = root_gauge_factory_proxy.ownership_admin()
+    default_emergency_admin = root_gauge_factory_proxy.emergency_admin()
+    assert root_gauge_factory_proxy.manager() == alice  # contract deployer is manager
+
+    # unauthorised users cannot change the manager:
+    with brownie.reverts():
+        root_gauge_factory_proxy.set_manager(charlie, {"from": bob})
+
+    # emergency admin and ownership admin can become manager:
+    for acct in [default_emergency_admin, default_ownership_admin]:
+        root_gauge_factory_proxy.set_manager(acct, {"from": acct})
+        assert root_gauge_factory_proxy.manager() == acct
+        chain.undo()
+
+    # authorised users can change the manager to anyone:
+    for acct in [default_emergency_admin, default_ownership_admin, alice]:
+        root_gauge_factory_proxy.set_manager(bob, {"from": acct})
+        assert root_gauge_factory_proxy.manager() == bob
+        chain.undo()
+
+
+def test_transfer_proxy_admins(root_gauge_factory_proxy, alice, bob, charlie):
+
+    default_ownership_admin = root_gauge_factory_proxy.ownership_admin()
+    default_emergency_admin = root_gauge_factory_proxy.emergency_admin()
+
+    # ---
+    # commit future proxy owners
+
+    # manager cannot commit new admins
+    with brownie.reverts():
+        root_gauge_factory_proxy.commit_set_admins(bob, charlie, {"from": alice})
+
+    # unauthorised accounts cannot commit new admins
+    with brownie.reverts():
+        root_gauge_factory_proxy.commit_set_admins(bob, bob, {"from": charlie})
+
+    # emergency admin cannot commit new admins
+    with brownie.reverts():
+        root_gauge_factory_proxy.commit_set_admins(bob, charlie, {"from": default_emergency_admin})
+
+    # commit future owners
+    root_gauge_factory_proxy.commit_set_admins(bob, charlie, {"from": default_ownership_admin})
+    assert root_gauge_factory_proxy.ownership_admin() == default_ownership_admin
+    assert root_gauge_factory_proxy.emergency_admin() == default_emergency_admin
+    assert root_gauge_factory_proxy.future_ownership_admin() == bob
+    assert root_gauge_factory_proxy.future_emergency_admin() == charlie
+
+    # ---
+    # accept new proxy owners
+
+    # unauthorised accts cannot accept future ownership and emergency admins on behalf:
+    for acct in [default_ownership_admin, default_emergency_admin, alice, charlie]:
+        with brownie.reverts():
+            root_gauge_factory_proxy.accept_set_admins({"from": acct})
+
+    root_gauge_factory_proxy.accept_set_admins({"from": bob})
+    assert root_gauge_factory_proxy.ownership_admin() == bob
+    assert root_gauge_factory_proxy.emergency_admin() == charlie
+
+
+def test_transfer_ownership(
```

Comment:
Can break this one up too, really into just the success cases
- Success commit transfer ownership of a factory
- Success accept transfer ownership of a factory
- Revert commit transfer ownership of a factory (proxy is unauthorized)
- Revert accept transfer ownership of a factory (proxy is unauthorized)